test/optim/test_optim.py

```diff
@@ -247,53 +246,6 @@ def fn_base(optimizer, weight, bias):
         for _i in range(20):
             optimizer.step(fn)
 
-        # Make sure that loading optimizers with step not wrapped in tensor can work
-        state_dict = optimizer.state_dict()
-        if "step" in state_dict["state"][0] and torch.is_tensor(
-            state_dict["state"][0]["step"]
-        ):
-            for state in state_dict["state"].values():
-                state["step"] = state["step"].item()
-            optimizer.load_state_dict(state_dict)
-            optimizer.step()
-
-        # Check that state dict can be loaded even when we cast parameters
-        # to a different type and move to a different device.
```

Comment:
So you may be wondering "okay Jane you've only migrated the cuda part, but not the casting part" and you would be correct! except if you look at the callsite of `_test_state_dict`, the things that are passed in _are_ already torch.float32 (default) so this test did unnecessary casting to itself.
https://github.com/pytorch/pytorch/blob/7e6a04e542ff4b1cd87dc2dba9095274604b26af/test/optim/test_optim.py#L331-L338

Moreover, changing the dtype would fail the test on just dtype comparison alone. If we find casting the params important to test, it should be a new test. This comment is just to clarify that we did not lose coverage.